lib/improver/metadata/utilities.py

```diff
@@ -31,10 +31,57 @@
 """General IMPROVER metadata utilities"""
 
 import hashlib
+import iris
 # Usage of pickle in this module is only for creation of pickles and hashing
 # the contents. There is no loading pickles which would create security risks.
 import pickle  # nosec
 
+import dask.array as da
+import numpy as np
+
+
+def create_new_diagnostic_cube(
+        name, units, coordinate_template, attributes=None, data=None,
+        datatype=np.float32):
+    """
+    Creates a template for a new diagnostic cube with suitable metadata.
+
+    Args:
+        name (str):
+            Standard or long name for output cube
+        units (str or cf_units.Unit):
+            Units for output cube
+        coordinate_template (iris.cube.Cube):
+            Cube from which to copy dimensional and auxiliary coordinates
+        attributes (dict or None):
+            Dictionary of attribute names and values
+        data (numpy.ndarray or None):
+            Data array.  If not set, cube is filled with zeros using a lazy
+            data object, as this will be overwritten later by the caller
+            routine.
+        datatype (numpy.dtype):
+            Datatype for dummy cube data if "data" argument is None.
+
+    Returns:
+        iris.cube.Cube:
+            Cube with correct metadata to accommodate new diagnostic field
+    """
+    if data is None:
+        data = da.zeros_like(coordinate_template, dtype=datatype)
```

Comment:
We need to use `core_data()` method, so that dask can pick up chunking if template has lazy data.
```suggestion
        data = da.zeros_like(coordinate_template.core_data(), dtype=dtype)
```

Reply:
OK